src/redux/uniqueTokens.ts

```diff
@@ -202,206 +168,40 @@ export const uniqueTokensResetState = () => (
-      shouldStopFetching =
-        newPageResults.length < UNIQUE_TOKENS_LIMIT_PER_PAGE ||
-        uniqueTokens.length >= UNIQUE_TOKENS_LIMIT_TOTAL;
-    } catch (error) {
-      dispatch({
-        showcase: !!showcaseAddress,
-        type: UNIQUE_TOKENS_GET_UNIQUE_TOKENS_FAILURE,
-      });
-      captureException(error);
-      // stop fetching if there is an error & dont save results
-      shouldStopFetching = true;
-      errorCheck = true;
-    }
-    return shouldStopFetching;
-  };
-
-  await fetchNetwork(currentNetwork);
-
-  // Only include poaps and L2 nft's on mainnet
-  if (currentNetwork === Network.mainnet) {
-    const poaps = (await fetchPoaps(accountAddress)) ?? [];
-    if (poaps.length > 0) {
-      analytics.identify(undefined, { poaps: poaps.length });
-      uniqueTokens = uniqueTokens.filter(token => token.familyName !== 'POAP');
-      uniqueTokens = uniqueTokens.concat(poaps);
-    }
-
-    // Fetch Optimism and Arbitrum NFTs
-    const layer2NFTs = await getNftsByWalletAddress(accountAddress);
-
-    if (layer2NFTs.length > 0) {
-      uniqueTokens = uniqueTokens.concat(layer2NFTs);
-    }
-
-    // we only care about analytics for mainnet + L2's
-    analytics.identify(undefined, { NFTs: uniqueTokens.length });
-
-    // Fetch recently registered ENS tokens (OpenSea doesn't recognize these for a while).
-    // We will fetch tokens registered in the past 48 hours to be safe.
-    const ensTokens = await fetchEnsTokens({
-      address: accountAddress,
-      timeAgo: { hours: 48 },
-    });
-    if (ensTokens.length > 0) {
-      uniqueTokens = uniqBy([...uniqueTokens, ...ensTokens], 'uniqueId');
-    }
-  }
-
-  // NFT Fetching clean up
-  // check that the account address to fetch for has not changed while fetching before updating state
-  const isCurrentAccountAddress =
-    accountAddress === (showcaseAddress || getState().settings.accountAddress);
-  if (!showcaseAddress && isCurrentAccountAddress && !errorCheck) {
-    saveUniqueTokens(uniqueTokens, accountAddress, currentNetwork);
-  }
-  if (isCurrentAccountAddress && !errorCheck) {
+  try {
+    const nfts = await fetchAllUniqueTokens(accountAddress, network);
     dispatch({
-      payload: uniqueTokens,
-      showcase: !!showcaseAddress,
+      payload: nfts,
+      showcase: false,
       type: UNIQUE_TOKENS_GET_UNIQUE_TOKENS_SUCCESS,
     });
-  }
-};
-
-/**
- * Revalidates a unique token via OpenSea API, updates state, and saves to local storage.
- *
- * Note:  it is intentional that there are no loading states dispatched in this action. This
- *        is for _revalidation_ purposes only.
- *
- * @param contractAddress - The contract address of the NFT
- * @param tokenId - The tokenId of the NFT
- * @param {Object} config - Optional configuration
- * @param {boolean} config.forceUpdate - Trigger a force update of metadata (equivalent to refreshing metadata in OpenSea)
- */
-export const revalidateUniqueToken = (
```

Comment:
removing this bc it is only used to revalidate unknown ENS nfts. I tested this out with a few different ens nfts that are "unknown" on OpenSea, and they were not an issue w/ simplehash. should definitely be covered in QA though